api/v1beta1/hetznerbaremetalhost_types.go

```diff
@@ -547,6 +548,9 @@ func (host *HetznerBareMetalHost) SetError(errType ErrorType, errMessage string)
 	}
 	host.Spec.Status.ErrorType = errType
 	host.Spec.Status.ErrorMessage = errMessage
+	if errType == PermanentError {
+		host.Annotations[PermanentErrorAnnotation] = time.Now().Format(time.RFC3339)
```

Comment:
is it worth using a utility or checking if annotations are empty?

Reply:
@janiskemper good point, I updated the code to check for nil. See 72fbd26f - ok now?